src/test/java/com/google/sps/CommentAnalysisTest.java

```diff
@@ -14,70 +14,99 @@
 
 package com.google.sps;
 
-import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
-import com.google.api.client.http.javanet.NetHttpTransport;
-import com.google.api.client.json.JsonFactory;
-import com.google.api.client.json.jackson2.JacksonFactory;
-import com.google.api.services.youtube.YouTube;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
 import com.google.api.services.youtube.model.CommentThreadListResponse;
 import com.google.sps.servlets.utils.CommentAnalysis;
+import com.google.sps.servlets.utils.Range;
 import com.google.sps.servlets.utils.Statistics;
-import java.io.IOException;
-import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Arrays;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 
-/** This is a JUnit test for sentiment analysis */
+/** This is a JUnit test for sentiment mockedAnalysis */
 @RunWith(JUnit4.class)
 public class CommentAnalysisTest {
-  private CommentThreadListResponse youtuberesponse;
-  private CommentAnalysis analysis;
-  private static final String APPLICATION_NAME = "testComment";
-  private static final String DEVELOPER_KEY = "API_KEY";
-  private static final String PLAINTEXT = "plainText";
-  private static final String SNIPPET = "snippet";
-  private static final String TEST_VIDEO_ID = "E_wKLOq-30M";
-  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
-
-  /**
-   * Build and return an authorized API client service.
-   *
-   * @return an authorized API client service
-   * @throws GeneralSecurityException, IOException
-   */
-  public static YouTube getService() throws GeneralSecurityException, IOException {
-    final NetHttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
-    return new YouTube.Builder(httpTransport, JSON_FACTORY, null)
-        .setApplicationName(APPLICATION_NAME)
-        .build();
-  }
+  private static final CommentAnalysis mockedAnalysis = mock(CommentAnalysis.class);
+  private static final CommentThreadListResponse mockedExpectedList =
+      mock(CommentThreadListResponse.class);
+  private static final CommentThreadListResponse mockedEdgeList =
+      mock(CommentThreadListResponse.class);
```

Comment:
Usually we should only mock objects that we are not able to instantiate normally during a test or whose behaviour we want to simulate. In practice, generally things like services, databases, or anything with an external dependency are mocked out.

The reason for this is that if we too agressively mock out objects under test, we may have them simulate behaviour that is not possible with real world objects.

E.g we could do something like:

```
when(mockDouble.doubleValue()).thenReturn(2);
when(mockDouble.equals(2)).thenReturn(false);
```

In this test, we shouldn't need to mock out `CommentThreadListResponse` objects.

Reply:
Since CommentThreadListResponse and CommentThread is also hard to initialize and simulate, to get a more well-covered test, I added the a mocked commentThreadList and mockedCommentThread. Please let me know if I misunderstood the anywhere. Thanks!

Reply:
I don't think it's a good idea to mock out `CommentThreadListResponse` objects and `CommentThread` objects since it will mean your unit tests will be operating on different classes of objects than what is actually being exercised in the code.

It shouldn't be too bad to create CommentThreadListResponse/CommentThread objects since they have explicit setters for the different fields that we're reading from.
[CommentThreadListResponse](https://developers.google.com/resources/api-libraries/documentation/youtube/v3/java/latest/com/google/api/services/youtube/model/CommentThreadListResponse.html)
[CommentThread](https://developers.google.com/resources/api-libraries/documentation/youtube/v3/java/latest/com/google/api/services/youtube/model/CommentThread.html)

Reply:
Updated by initialize the comment, commentSnippet, commentThead, commentThreadListResponse and commentThreadSnippet